tests/repositories/test_legacy_repository.py

```diff
@@ -309,16 +309,16 @@ def test_get_200_returns_page(http):
     assert repo._get("/foo")
 
 
-def test_get_404_returns_none(http):
-    repo = MockHttpRepository({"/foo": 404}, http)
+def test_get_40x_and_returns_none(http):
```

Comment:
You should also maybe consider using [pytest.parameterize](https://docs.pytest.org/en/stable/parametrize.html) to pass in the different status codes, this means that each case would be an independent case rather than a bulk test failure if things regress.

Reply:
Hello @abn, done rebasing/fixing conflicts, and updating tests with parameterize. 😸

Reply:
Oh hey! sorry i have been terrible at coming back to this. See @stephsamson comment :) Again, apologies.